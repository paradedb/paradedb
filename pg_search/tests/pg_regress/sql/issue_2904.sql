\i common/common_setup.sql

DROP TABLE IF EXISTS test;
CREATE TABLE test (id serial primary key, description text) PARTITION BY RANGE (id);

CREATE TABLE test_p1 PARTITION OF test FOR VALUES FROM (1) TO (5);
CREATE TABLE test_p2 PARTITION OF test FOR VALUES FROM (5) TO (10);

INSERT INTO test(description)
VALUES ('hello');

CREATE INDEX test_idx ON test USING bm25 (id, description) WITH (key_field = 'id');

SELECT * FROM test WHERE description @@@ 'hello';
DROP TABLE test;
