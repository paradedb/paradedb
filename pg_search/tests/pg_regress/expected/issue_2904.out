\i common/common_setup.sql
CREATE EXTENSION IF NOT EXISTS pg_search;
-- Disable parallel workers to avoid differences in plans
SET max_parallel_workers_per_gather = 0;
SET enable_indexscan to OFF;
SET paradedb.enable_mixed_fast_field_exec = true;
DROP TABLE IF EXISTS test;
CREATE TABLE test (id serial primary key, description text) PARTITION BY RANGE (id);
CREATE TABLE test_p1 PARTITION OF test FOR VALUES FROM (1) TO (5);
CREATE TABLE test_p2 PARTITION OF test FOR VALUES FROM (5) TO (10);
INSERT INTO test(description)
VALUES ('hello');
CREATE INDEX test_idx ON test USING bm25 (id, description) WITH (key_field = 'id');
SELECT * FROM test WHERE description @@@ 'hello';
 id | description 
----+-------------
  1 | hello
(1 row)

DROP TABLE test;
